packages/ckeditor5-engine/src/model/differ.ts

```diff
@@ -1520,7 +1520,17 @@ function _generateDiffInstructionsFromChanges( oldChildrenLength: number, change
 			// We removed `howMany` old nodes, update `oldChildrenHandled`.
 			oldChildrenHandled += change.howMany;
 		} else {
-			diff.push( ...'a'.repeat( change.howMany ).split( '' ) );
+			// Total maximum amount of arguments that can be passed to `Array.prototype.push` may be limited so we need to
+			// add them manually one by one to avoid this limit. However loop might be a bit slower than `push` method on
+			// smaller changesets so we need to decide which method to use based on the size of the change.
+			// See: https://github.com/ckeditor/ckeditor5/issues/16819
+			if ( change.howMany > 1500 ) {
```

Comment:
Could you write (here and/or in the issue) why we choose 1500?

It's based on some tests or it's some kind of random value?

Reply:
Limitation of args stack depends on OS / Browser etc. I chose some kind of random value.